src/key.js

```diff
@@ -359,12 +385,22 @@ class Key {
   /**
    * Concats one or more Keys into one new Key.
    *
-   * @param {Array<Key>} keys
+   * @param {Array<Key>} keys - The array of keys to concatenate
    * @returns {Key}
    */
   concat (...keys) {
     return Key.withNamespaces([...this.namespaces(), ...flatten(keys.map(key => key.namespaces()))])
   }
+
+  /**
+   * Check if value is a Key instance
+   *
+   * @param {any} value - Value to check
+   * @returns {boolean}
```

Comment:
```suggestion
   * @returns {value is Key}
```

This would enable following to type check:

```ts
if (Key.isKey(x)) {
  x.child(y)
}
```

Without this change it would not be able to refine `x` type to `Key`